multicluster/controllers/multicluster/leader/clusterset_controller.go

```diff
@@ -205,8 +205,15 @@ func (r *LeaderClusterSetReconciler) updateStatus() {
 		// from ClusterClaim to update the CR, otherwise, the update will fail due
 		// to invalid ClusterID.
 		clusterSet.Spec.ClusterID = string(r.clusterID)
+		err = r.Update(context.TODO(), clusterSet)
+		if err != nil {
+			klog.ErrorS(err, "Failed to update ClusterSet's ClusterID", "name", namespacedName)
+		}
+		// Status will be updated in next interval after the ClusterSet spec is updated.
```

Comment:
Is Status updated periodically? Why?

Why we cannot update Status right after the spec update?

Do we have a better place to update the spec with ClusterID, like when the controller first sees the ClusterSet?